sqlalchemy_ibmi/base.py

```diff
@@ -753,15 +753,12 @@ def _parse_dbapi_version(self, vers):
         return vers
 
     def _get_server_version_info(self, connection, allow_chars=True):
-        # NOTE: this function is not reliable, particularly when
-        # freetds is in use.   Implement database-specific server version
-        # queries.
-        dbapi_con = connection.connection
-        version = []
-        r = re.compile(r"[.\-]")
-        for n in r.split(dbapi_con.getinfo(self.dbapi.SQL_DBMS_VER)):
-            version.append(int(n))
-        return tuple(version)
+        if getattr(connection, "connection"):
```

Comment:
Why the getattr here? Is there a way that it would not have a connection attribute?

This is called by [DefaultDialect.initialize](https://github.com/sqlalchemy/sqlalchemy/blob/fa1f67a01e80367d73cf5d1d93b6f7f51dc1746b/lib/sqlalchemy/engine/default.py#L307). The [docs]( https://docs.sqlalchemy.org/en/13/core/internals.html#sqlalchemy.engine.default.DefaultDialect.initialize) say it should have a connection object with full capabilities and I don't see any other dialect checking that the connection is there.

Reply:
Saw it in set_isolation_level, but I can remove it from that too when I refactor it for the pyodbc isolation levels

Reply:
Sounds good. Let's remove this one in this PR and then when you do the isolation levels remove the other one as you said.